src/gretel_client/transformers/transformers/bucket.py

```diff
@@ -1,34 +1,50 @@
 from dataclasses import dataclass
-from numbers import Number
 from typing import Union, List, Optional, Tuple
 
 from gretel_client.transformers.base import TransformerConfig, Transformer
 
 
-class BucketError(Exception):
-    pass
-
-
-class BucketRange:
-    def __init__(self, buckets: List[Tuple], *, labels: Union[List[Number], List[str]] = None, min_label=None,
-                 max_label=None,
-                 auto_label_prefix='BUCKET_'):
-        # buckets should be all strings or all numbers
-        str_check = [isinstance(item[0], str) and isinstance(item[1], str) for item in buckets]
-        num_check = [isinstance(item[0], Number) and isinstance(item[1], Number) for item in buckets]
-        if not all(str_check) and not all(num_check):
-            raise BucketError('buckets must be all strings or all numbers')
-        self.buckets = buckets
-        if not labels:
-            labels = [auto_label_prefix + str(x) for x in range(len(buckets))]
-        self.labels = labels
-        self.min_bucket = min_label if min_label is not None else labels[0]
-        self.max_bucket = max_label if max_label is not None else labels[-1]
-
-
 @dataclass(frozen=True)
 class BucketConfig(TransformerConfig):
-    bucket_range: BucketRange = None
+    buckets: Union[List[float], List[int], Tuple[Union[float, int], Union[float, int], Union[float, int]]] = None
+    bucket_labels: Union[List[float], List[int], List[str]] = None
+    lower_outlier_label: Union[float, int, str] = None
+    upper_outlier_label: Union[float, int, str] = None
```

Comment:
Can we have some doc string here that detail the params, etc? It's not clear to me what the valid `buckets` options are, etc. We'll want some explanation of what's happening under the hood I think